content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t1 register.
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t2 register, and will take over t3+.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 5:
+This case shows how to declare one Unbounded Array for each resource type. The rule of thumb is to always declare each Unbounded Array after the last non-Unbounded resource.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV_bounded[3];  // Will be bound to t1 register... Until t3 (inclusive).
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t4 register, and will take over t5+.
+        RWTexture2D<float4>      m_texUAV_bounded[5];  // Will be bound to u0 register... Until u4 (inclusive).
+    	RWTexture2D<float4>      m_texUAV_unbounded[]; // Will be bound to u5 register, and will take over u6+.
+        Sampler                  m_sampler1_bounded[7]; // Will be bound to s0 register... Until s6 (inclusive).
+        Sampler                  m_sampler2_bounded[3]; // Will be bound to s7 register... Until s9 (inclusive).
+        Sampler                  m_sampler_unbounded[]; // Will be bound to s10 register, and will take over s11+.
+        ConstantBuffer<MyStruct> m_cb_array[2]; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register)... Until     b2 (inclusive).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b3 register.
+        ConstantBuffer<MyStruct> m_cb_unbounded[]; // Will be bound to b4 register, and will take over b5+.
+    };
```

Comment:
```suggestion
ShaderResourceGroupSemantic slot1
{
    FrequencyId = 1;
};

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};

//For DX12, --unique-idx is OFF, so the expected register assignment is as follows
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV1;             // Binds to t0 register.
    Texture2D<float4>        m_texSRV_bounded[3];   // Binds from t1 to t3 (inclusive).
    Texture2D<float4>        m_texSRV_unbounded[];  // Binds to t4 register and takes over t5+.
    RWTexture2D<float4>      m_texUAV_bounded[5];   // Binds from u0 to u4 (inclusive).
    RWTexture2D<float4>      m_texUAV_unbounded[];  // Binds to u5 register and takes over u6+.
    Sampler                  m_sampler1_bounded[7]; // Binds from s0 to s6 (inclusive).
    Sampler                  m_sampler2_bounded[3]; // Binds from s7 to s9 (inclusive).
    Sampler                  m_sampler_unbounded[]; // Binds to s10 register and takes over s11+.
    ConstantBuffer<MyStruct> m_cb_array[2];         // Binds to b1 register. (The ConstantBuffer reserved for SRG1 binds to b0 register).
    ConstantBuffer<MyStruct> m_cb2;                 // Binds to b3 register.
    ConstantBuffer<MyStruct> m_cb_unbounded[];      // Binds to b4 register, and takes over b5+.
};
```
* Moved comment: "(Because the ConstantBuffer reserved for….)" to simplify the comments for each resource declaration. 
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".
* Simplified "will take over" to "takes over"

Reply:
Fixed.